### PR TITLE
Fix ability to use patch on Windows CI machines

### PR DESCRIPTION
**Pull request by skottmckay**
### Description

Add 32-bit patch binary and infra to fallback to it. The Azure devops Windows CIs are missing patch.exe from their git install for some reason so the default `find_package(Patch)` fails as that is where it expects to find it.

Remove Eigen patch. Underlying issue was fixed in source 3 years ago by https://gitlab.com/libeigen/eigen/-/commit/c6c84ed9611cafd0acf89e467b050d9e94bec686 and the patch command is invalid (args are for git apply not patch).

### Motivation and Context

Make usage of patch consistent across all CIs
Fix https://github.com/microsoft/onnxruntime/issues/15248

### CODE REVIEW

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -526,7 +526,21 @@ if(NOT WIN32 AND NOT CMAKE_SYSTEM_NAME STREQUAL "Android")
   find_package(Iconv REQUIRED)
   set(ICONV_LIB Iconv::Iconv)
 endif()
+
 find_package(Patch)
+if (WIN32 AND NOT Patch_FOUND)
+    # work around CI machines missing patch from the git install by falling back to the binary in this repo.
+    # replicate what happens in https://github.com/Kitware/CMake/blob/master/Modules/FindPatch.cmake but without
+    # the hardcoded suffixes in the path to the patch binary.
+    find_program(Patch_EXECUTABLE NAMES patch PATHS ${PROJECT_SOURCE_DIR}/external/git.Win32.2.41.03.patch)
```
Comment (snnn) [inline cmake/CMakeLists.txt:535]: But, does this binary work fine on Windows ARM64 machines?

Comment (skottmckay) [reply cmake/CMakeLists.txt:535]: in theory: https://learn.microsoft.com/en-us/windows/arm/overview#support-for-existing-windows-apps-on-arm. 

I'll test something on the QNN ARM64 CI to check

**Review by snnn — APPROVED** [review]

File: cmake/CMakeLists.txt (hunk 1 above)
Comment (skottmckay) [reply cmake/CMakeLists.txt:535]: Seems to work. https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1208745&view=logs&j=ca395085-040a-526b-2ce8-bdc85f692774&t=cb33ee24-f492-54d2-727a-ae0a7bb84b2c is from running the patch.exe with `--version` and `--help` on the QNN ARM64 CI.

Comment (snnn) [reply cmake/CMakeLists.txt:535]: Thank you!

**Pull request merged**